interfaces/builtin/personal_files.go

```diff
@@ -76,3 +80,62 @@ func init() {
 		},
 	})
 }
+
+// potentiallyMissingDirs returns an ensure directory specification that contains the information
+// required to create potentially missing directories for the given raw path. Potentially missing
+// directories are those that are implicitly required in order to enable the user to create
+// the target directory specified in a write attribute path.
+func potentiallyMissingDirs(rawPath interface{}) (*interfaces.EnsureDirSpec, error) {
+	path, ok := rawPath.(string)
+	if !ok {
+		// BeforePreparePlug should prevent this
+		return nil, fmt.Errorf("%[1]v (%[1]T) is not a string", rawPath)
+	}
+
+	// All directories between $HOME and the leaf directory are potentially missing
+	path = filepath.Clean(path)
+	pathElements := strings.Split(path, string(filepath.Separator))
+	if pathElements[0] != "$HOME" {
+		// BeforePreparePlug should prevent this
+		return nil, fmt.Errorf(`%q must start with "$HOME/"`, path)
+	}
+	if len(pathElements) < 3 {
+		return nil, nil
+	}
+
+	return &interfaces.EnsureDirSpec{
+		// EnsureDir prefix directory that must exist
+		MustExistDir: pathElements[0],
+		// Directory to ensure by creating the missing directories within MustExistDir
+		EnsureDir: filepath.Join(pathElements[:len(pathElements)-1]...),
+	}, nil
+}
+
+func dirsToEnsure(rawPaths []interface{}) ([]*interfaces.EnsureDirSpec, error) {
+	var ensureDirSpecs []*interfaces.EnsureDirSpec
+	for _, rawPath := range rawPaths {
+		ensureDirSpec, err := potentiallyMissingDirs(rawPath)
+		if err != nil {
+			return nil, err
+		}
+		if ensureDirSpec != nil {
+			ensureDirSpecs = append(ensureDirSpecs, ensureDirSpec)
+		}
+	}
+	return ensureDirSpecs, nil
+}
+
+func (iface *personalFilesInterface) MountConnectedPlug(spec *mount.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	// Create missing directories for write paths only
+	var writes []interface{}
+	_ = plug.Attr("write", &writes)
```

Comment:
we should check the error, see stringListAttribute, maybe we can even reuse it and move into utils.go?

Reply:
shared memory interface defined `stringListAttribute`. It checks the error in `beforePrepare`, but not when using it thereafter.

I moved `stringListAttribute` into `interfaces/helpers.go`, cannot use utils due to  circular dependency. Let me know if there is a better place to put it, perhaps `connection.go`?

Reply:
I'm confused about the remark on circular dependecy, I meant a file in this package. interfaces/builtin/utils.go

Reply:
Followed you proposal, thanks.

There are use of splot/plug.Atrr(...) outside of builtin which in theory can also adopt the use of StringListAttribute(), then having it in builtin did not feel accurate, but putting it in interfaces/utils/ introduce circular dependency.

But I think having it in builtin is nice and simple for now. There are however many places where arguable this function should be adopted in favour of passing []interface{} into validation function that only later validate it's a string list, but this feels outside of current scope.